src/main/resources/Confluence/Macros/MicrosoftStream.xml

```diff
@@ -0,0 +1,1039 @@
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity output="false"}}
+#macro (getAlignmentClass $alignment $alignmentClass)
+  #if ($alignment == 'left')
+    #set ($alignmentClass = 'pull-left')
+  #elseif ($alignment == 'right')
+    #set ($alignmentClass = 'pull-right')
+  #elseif ($alignment == 'center')
+    #set ($alignmentClass = 'microsoftStreamMacroCenter')
+  #end
+#end
+
+## Extract the base URL and query string parameters.
+#macro (getBaseURLAndQueryParams $url $baseURL $queryParams)
+  #set ($baseURL = $url)
+  ## Use $urltool.toURL after XWiki parent version is &gt;= 12.9RC1 to include XCOMMONS-2042: Add a method in URLTool
+  ## to convert a string in an URL object.
+  #if ($stringtool.contains($url, '?'))
+    #set ($baseURL = $stringtool.split($url, '?')[0])
+    #set ($queryString = $stringtool.split($url, '?')[1])
+  #end
+  ## Depending on how the link was shared, it could be missing this part.
+  #if (!$stringtool.contains($baseURL, 'embed/video'))
+    #set ($baseURL = $stringtool.replace($baseURL, 'video', 'embed/video'))
+  #end
+  #set ($urlParams = {})
+  #if ($!queryString)
+    #set ($queryParams = $urltool.parseQuery($queryString))
+  #end
+#end
+
+#macro (getStartAsSeconds $start $startSeconds)
+  #if ($datetool.toDate('HH:mm:ss', $start))
+    #set ($dateObj = $datetool.toDate('HH:mm:ss', $start))
+  #elseif ($datetool.toDate('mm:ss', $start))
+    #set ($dateObj = $datetool.toDate('mm:ss', $start))
+  #elseif ($datetool.toDate('ss', $start))
+    #set ($dateObj = $datetool.toDate('ss', $start))
+  #end
+  #set ($startSeconds = 0)
+  #if ($!dateObj)
+    #set ($startSeconds = 3600 * $dateObj.getHours() + 60 * $dateObj.getMinutes() + $dateObj.getSeconds())
+  #end
+#end
+{{/velocity}}
+
+{{velocity}}
+#set ($discard = $xwiki.ssx.use('Confluence.Macros.MicrosoftStream'))
+#getAlignmentClass($stringtool.lowerCase($xcontext.macro.params.alignment) $alignmentClass)
+#getBaseURLAndQueryParams($xcontext.macro.params.url $baseURL $queryParams)
+## Update the query params, if needed, since macro parameters have priority over the parameters extracted from URL.
+#set ($autoplay = $xcontext.macro.params.autoplay)
+#if ($!autoplay)
+  #set ($discard = $queryParams.put('autoplay', $autoplay))
+#elseif (!$!autoplay &amp;&amp; !$queryParams.autoplay)
+  #set ($discard = $queryParams.put('autoplay', false))
+#end
+#set ($showinfo = $xcontext.macro.params.showinfo)
+#if ($!showinfo)
+  #set ($discard = $queryParams.put('showinfo', $showinfo))
+#elseif (!$!showinfo &amp;&amp; !$queryParams.showinfo)
+  #set ($discard = $queryParams.put('showinfo', false))
```

Comment:
`showInfo` is `true` by default on the Microsoft stream side.
But I think it would be good to use their default value here as well, since it's the value the user might expect.